hooks/tk-multi-publish2/nuke/publish_files_comp.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import nuke
+import sgtk
+import itertools
+from dd.runtime import api
+api.load("frangetools")
+import frangetools
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class NukePublishFilesDDCompValidationPlugin(HookBaseClass):
+    """
+    Inherits from NukePublishFilesPlugin
+    """
+    def _build_dict(self, seq, key):
+        """
+        Creating a dictionary based on a key.
+
+        :param seq: list of dictionaries
+        :param key: dictionary key from which to create the dictionary
+        :return: dict with information for that particular key
+        """
+        return dict((d[key], dict(d, index=index)) for (index, d) in enumerate(seq))
+
+
+    def _framerange_to_be_published(self, item):
+        """
+        Since users have the option to render only a subset of frames,
+        adding validation to check if the full frame range is being published.
+
+        :param item: Item to process
+        :return: True if yes false otherwise
+        """
+        lss_path = item.properties['node']['cached_path'].value()
```

Comment:
You already have this cached in item.properties["sequence_paths"] so no need to calculate again. The code should be:

```
if item.properties["is_sequence"]:
    files = item.properties["sequence_paths"]
else:
    files = item.properties["path"]
```

Reply:
Noticed that if sequence, then also we have the path stored in item.properties["path"].
so keeping lss_path = item.properties['path']